src/_paths.ts

```diff
@@ -0,0 +1,78 @@
+import { Key } from './_types';
```

Comment:
I put all this in a separate file because I imagine it could be used to update `pathOr` to support an arbitrary path length (instead of just 3 path segments) and path strings.